### ISSUE TITLE
Virtual machines are turned on before they are done building

### ISSUE THREAD

**Issue by jeffcpullen** [ahuffman/sat6-create-hosts#4]
The block that actually creates the VMs doesn't have a check between building and powering on. If the VM hasn't finished building it errors out.

A hack of adding in a pause there works. A better solution would be a check to see if the VM has finished building. Not sure if this can be done from the Satellite API, or if it would have to come from the infrastructure.

**Comment by ahuffman** [comment]
Hi @jeffcpullen,

I'm testing a pause in there.  I use libvirt as  a compute resource in my lab, and intermittently have problems where the pxe boot loader doesn't pick up the boot image.  Typically if I restart the newly created VM, it properly picks up the pxe image.  It seems to be something with the libvirt networking drivers.  

As far as the immediate power-on command that the code does, this is exactly what the Satellite6 UI does when creating a new VM over a compute resource.  As soon as the host is created it generates a power-on command over the compute resource.  I'm wondering if you're having a similar pxe boot issue as I have with libvirt in my lab.  

Are you able to share a little more about the environment you're testing this role in? 

-Andrew

**Comment by jeffcpullen** [comment]
I'm not sure about the expected behavior. I could probably test to see if Satellite powers on the host once it is finished building. I can say that with my environment it takes ~20 seconds for the VM to leave the "locked" state when being created. 

### Error Message
```{
    "x_runtime": "2.097651",
    "status": 500,
    "set_cookie": "request_method=PUT; path=/; secure; HttpOnly; SameSite=Lax, _session_id=a0894dddcc3a83c369967482aec0d4cc; path=/; secure; HttpOnly; SameSite=Lax",
    "x_powered_by": "Phusion Passenger 4.0.18",
    "x_content_type_options": "nosniff",
    "x_request_id": "8bb2bb62-efb9-44e0-86b4-d03322a11102",
    "content": "{\n  \"error\": {\"message\":\"Ovirt client returned an error: Cannot run VM: VM is locked. Please try again in a few minutes.\"}\n}\n",
    "x_permitted_cross_domain_policies": "none",
    "json": {
        "error": {
            "message": "Ovirt client returned an error: Cannot run VM: VM is locked. Please try again in a few minutes."
        }
    },
    "invocation": {
        "module_args": {
            "directory_mode": null,
            "force": false,
            "remote_src": null,
            "status_code": [
                "200"
            ],
            "follow": false,
            "owner": null,
            "body_format": "json",
            "client_key": null,
            "group": null,
            "use_proxy": true,
            "unsafe_writes": null,
            "serole": null,
            "content": null,
            "setype": null,
            "follow_redirects": "safe",
            "return_content": false,
            "method": "PUT",
            "body": {
                "power_action": "start"
            },
            "url_username": "admin",
            "src": null,
            "dest": null,
            "selevel": null,
            "force_basic_auth": true,
            "removes": null,
            "http_agent": "ansible-httpget",
            "user": "admin",
            "regexp": null,
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "url_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "url": "https://sat6.homelab.hidden/api/v2/hosts/53/power",
            "backup": null,
            "seuser": null,
            "client_cert": null,
            "creates": null,
            "headers": {
                "Content-Type": "application/json"
            },
            "delimiter": null,
            "mode": null,
            "timeout": 300,
            "attributes": null,
            "validate_certs": true
        }
    },
```

### Playbook
```---
- hosts: rhv-ocp-ansible.homelab.hidden
  vars_files:
    - vault.yml
  gather_facts: no
  become: yes
  tasks:

  - name: "Provision Satellite 6 Hosts"
    include_role:
      name: "sat6-create-hosts"
    vars:
      sat6_hosts:
        #VM
        - name: "{{ item.name }}"
          host_group: "ose-rhel-7-5"
          location: "leier"
          ipv4: "{{ item.ipv4 }}"
          domain: "homelab.hidden"
          comment: "{{ item.comment }}"
          compute_resource: "rhv.homelab.hidden"
          compute_profile: "ose-node"
          force_host_update: False
          manage_host: True
    with_items:
      - { name: rhv-ocp-master-1, ipv4: 192.168.2.221, comment: "OCP Master 1" }
      - { name: rhv-ocp-infra-1, ipv4: 192.168.2.222, comment: "OCP Infra 1" }
      - { name: rhv-ocp-app-1, ipv4: 192.168.2.223, comment: "OCP App 1" }
```

**Comment by ahuffman** [comment]
Thanks @jeffcpullen, it does look like you are hitting a race condition.  I'll leave in the new task that offers a configurable pause second length prior to powering on the new VMs to account for this, but I personally am not hitting it with libvirt as a compute resource.  I'm still trying to determine why in my lab environment I tend to have intermittent issues with the pxeboot loader not picking up a boot image, but it does not seem to be related to pausing prior to a power on in my case.  I just need to update the docs on the new option for pausing, and I'll ship a new release.  Thanks for reporting your issue, as it's difficult to test for many compute resources in a limited lab, appreciate it!
